tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -201,27 +202,71 @@ private static void TryAddEndPoint(Span span)
             }
         }
 
+        private static bool AreArchitectureAndOsSupported()
+        {
+            var frameworkDescription = FrameworkDescription.Instance;
+            var osSupported = false;
+            var supportedOs = new[]
+            {
+                OSPlatform.Linux,
+                OSPlatform.MacOS,
+                OSPlatform.Windows
+            };
+            if (supportedOs.Contains(frameworkDescription.OSPlatform))
+            {
+                osSupported = true;
+            }
+
+            var archSupported = false;
+            var supportedArchs = new[]
+            {
+                ProcessArchitecture.Arm,
+                ProcessArchitecture.X64,
+                ProcessArchitecture.X86
+            };
+            if (supportedArchs.Contains(frameworkDescription.ProcessArchitecture))
+            {
+                archSupported = true;
+            }
+
+            if (!osSupported || !archSupported)
+            {
+                Log.Error(
+                    "AppSec could not start because the current environment is not supported. No security activities will be collected. Please contact support at https://docs.datadoghq.com/help/ for help. Host information: operating_system: {{ {OSPlatform} }}, arch: {{ {ProcessArchitecture} }}, runtime_infos: {{ {ProductVersion} }}",
+                    frameworkDescription.OSPlatform,
+                    frameworkDescription.ProcessArchitecture,
+                    frameworkDescription.ProductVersion);
+            }
+
+            return osSupported && archSupported;
+        }
+
         /// <summary>
-        /// Frees resouces
+        /// Frees resources
         /// </summary>
         public void Dispose() => _waf?.Dispose();
 
-        private void Report(ITransport transport, Span span, WafMatch[] results, bool blocked)
+        private void InstrumentationGateway_AddHeadersResponseTags(object sender, InstrumentationGatewayEventArgs e)
+        {
+            if (e.RelatedSpan.GetTag(Tags.Origin) == "appsec")
```

Comment:
I think I would prefer the Tag.AppSecEvent == true, but it's not a big deal, we could consider changing later.

Reply:
fixed in d6078e6aae7a16e96e83f9f38cb03687ee837b13